### ISSUE TITLE
Breaking linting setup

### ISSUE THREAD

**Issue by webbertakken** [game-ci/unity-builder#356]
Just  installed unity-builder repo on a new pc. I found that it's installing 4 additional hooks and tests are running both through lint staged (only the relevant ones)  and as part of the git hook (all of them). 

Subsequently
- Lint-staged no longer fails
- The catch all test that's now in the pre-commit hook has a bunch of tests that ended up failing (on windows) too 
 
Result:
It looks like the repository is currently not very developer friendly.

I'm happy with the upgrades to the latest husky and lint-staged on all repos but now we have to make sure that:
1. we only lint staged files on pre-commit (i.e. using lint staged) to minimize work inside pre-commit hook
2. it should work on Linux, MacOS and Windows
3. a fresh clone should require no additional steps other than running `yarn` and having IDE plugins installed 
4. no new tracked files should be generated right after cloning or running yarn.

**Comment by davidmfinol** [comment]
I think the changes we've made for unity-builder seem to be working well, so is the next step simply to apply the same changes to the other repos?

**Comment by webbertakken** [comment]
Yes. One more thing we should also do is upgrade the requirements to node >=16, as it allows for use of `fs/promises` and resolves some other weird bugs.
